games/static/games/proto.js

```diff
@@ -307,87 +347,146 @@ class Board {
+        this.squareAt(nextRocketLocation).setCharacter(alienHit)
+    }
+    reduceAlienHealthByOneAt(nextRocketLocation) {
+        this.squareAt(nextRocketLocation).character.health -= 1
+    }
+    waitAndEraseRocketAt(rocketLocation) {
+        setTimeout(() => { this.liftRocketAt(rocketLocation) }, 100)
+    }
+    theTopIs(rocketLocation) {
+        return this.squareAt(rocketLocation).isTop()
+    }
+    update(nextRocketLocation) {
+        return nextRocketLocation + this.directionUp()
+    }
+    updateIndexOf(rocket) {
+        rocket.index = rocket.index + this.directionUp()
+    }
+    liftRocketAt(rocketLocation) {
+        return this.squareAt(rocketLocation).pickCharacterUp()
+    }
+    getNextIndexOf(rocket) {
+        return rocket.index + this.directionUp()
+    }
+    getIndexOf(rocket) {
+        return rocket.index
+    }
+    createRocket() {
+        return new Rocket('rocket', this.squareAboveDefender(), this)
+    }
+    
+    squareAboveDefender() {
+        return this.defenderLocation - this.width
+    }
+
+    squareAt(rocketLocation) {
+        return this.squares[rocketLocation]
+    }
+
+    directionUp() {
+        return -this.width
+    }
+    
+}
+function calcNextAlienMove(previousDirection, counter, numberOfMoves, down) {
+    if (counter < numberOfMoves) {
+        return {
+            'counter': increaseByOne(counter),
+            'previousDirection':  previousDirection, 
+            'direction': previousDirection,
+            
         }
-        switch(e.keyCode ) {
-            case 32:
-                rocketId = setInterval(moveRocket , 100 , rocket)
-                break
+    }
+    if (counter === numberOfMoves) {
+        return {
+            'counter': zero(counter),
+            'previousDirection': getTheOppositeOf(previousDirection),
+            'direction': down,
+            
         }
-        
     }
+}
 
-    
+function increaseByOne(counter){
+    return counter += 1
 }
 
+function zero(counter) {
+    return counter = 0
+}
+
+function getTheOppositeOf(direction) {
+    return direction * -1
+}
 
 document.addEventListener('DOMContentLoaded', () => {
-    let grid = document.querySelector('.grid')
+    // let grid = document.querySelector('.grid')
     let alienInvaders = {
-        0:1, 1:1, 2:1, 3:1, 4:1, 5:1, 6:1, 7:1, 8:1, 9:1,
-        15:1, 16:2, 17:2, 18:2, 19:2, 20:2, 21:2, 22:2, 23:2, 24:2,
-        30:1, 31:1, 32:1, 33:2, 34:1, 35:1, 36:1, 37:1, 38:1, 39:1,
+        0: 2, 1: 1, 2: 1, 3: 1, 4: 1, 5: 1, 6: 1, 7: 1, 8: 1, 9: 1,
+        15: 2, 16: 1, 17: 1, 18: 1, 19: 1, 20: 1, 21: 1, 22: 1, 23: 1, 24: 1,
+        30: 2, 31: 1, 32: 1, 33: 1, 34: 1, 35: 1, 36: 1, 37: 1, 38: 1, 39: 1,
     }
-    setup = new Setup(alienInvaders, [], [])
-    board = new Board(grid, 15, setup)
-    board.draw(setup)
+    setup = new Setup(alienInvaders, [])
+    
     const startButton = document.querySelector('#start')
```

Comment:
why is this outside the game class?

Reply:
this was just for testing purposes

Reply:
i mean the start button - should be inside with the rest
its also makes sense that the game class is responsible for the start button

Reply:
its not there anymore